### ISSUE TITLE
Outbound Endpoint Health Check unable to serialize PingMessage when using Avro Serializer

### ISSUE THREAD

**Issue by cookie-bytes** [BEagle1984/silverback#201]
Hi

When using `SerializeAsAvro` and using `AddOutboundEndpointsCheck()` you get the following serialization exception:

_System.InvalidCastException: Unable to cast object of type 'Silverback.Messaging.HealthChecks.PingMessage' to type 'Some.Business.Event.Type'_

**Comment by cookie-bytes** [comment]
Could the PingMessage be added to Core or does this need to stay in the HealthChecks namespace and package?

**Comment by BEagle1984** [comment]
I have to think about this because I'd gladly avoid bringing a dependency on Avro into the other packages.

I honestly don't know if it's worth investing in this health check since it proved to be impractical in many cases as it just clutters the topic and is detrimental to interoperability.

**Comment by cookie-bytes** [comment]
This health check verifies the presence of all topic endpoints and the service’s ability to publish events to them using a Ping message. 
If the outbox worker is used, I suppose the outbox health check can be relied upon instead of the outbound endpoint health check. 
For services not using the outbox worker, error logs can be checked for producer error messages and this can be used to stop\restart the service if necessary.
Does that sound about right?

**Comment by BEagle1984** [comment]
Yes, exactly. For produce errors, we rely on the exception being thrown by the `Publish` / `Produce` methods. The caller can log it and monitor the logs to raise an alarm.
The downside is that you don't notice the issue until you try to produce and this might be a problem if you have for instance a service that only produces at night or during the weekend.

But having these ping messages polluting the topic is pretty ugly.
